lua/Network.lua

```diff
@@ -0,0 +1,651 @@
+local function getKey(line)
+	local keys = {k1, k2}
+	local bePoint = getBELine(line)
+
+	keys.P1 = line.OBJET_ID_8.."P1"
+	keys.P2 = line.OBJET_ID_8.."P2"
+
+	return keys
+end
+
+local function distanceRoute(target, line, vecDistance, fsPointers)
+	local distanceP = {
+		firstC,
+		secondC
+	}
+	local bePoint1 = getBELine(target)
+	local bePoint2 = getBELine(line)
+	local distance = fsPointers.firstP:distance(fsPointers.secondP)
+
+	if bePoint1.p1:contains(fsPointers.firstP) then
+		vecDistance[line.OBJET_ID_8.."P1"] = vecDistance[target.OBJET_ID_8.."P1"]
+		vecDistance[line.OBJET_ID_8.."P2"] = (distance  + vecDistance[target.OBJET_ID_8.."P2"])
+	else
+		vecDistance[line.OBJET_ID_8.."P2"] = vecDistance[target.OBJET_ID_8.."P2"]
+		vecDistance[line.OBJET_ID_8.."P1"] = (distance  + vecDistance[target.OBJET_ID_8.."P1"])
+	end
+    
+	return getKey(line)
+end
+
+local function checkLinePoints(line, target)
+	local points = {
+		firstP,
+		secondP
+	}
+	local bePoint = getBELine(line)
+	local dc1 = bePoint.p1:distance(target)
+	local dc2 = bePoint.p2:distance(target)
+
+	if dc1 > dc2 then
+		points.firstP = bePoint.p1
+		points.secondP = bePoint.p2
+	else
+		points.firstP = bePoint.p2
+		points.secondP = bePoint.p1
+	end
+
+	return points
+end
+
+local function getNextRoute(line, loopLine)
+	local countLine = 1
+
+	while line.route[countLine] do
+		if hasValue(loopLine, line.route[countLine]) then
+			table.insert(loopLine, line.route[countLine])
+		end
+
+		countLine = countLine + 1
+	end
+end
+
+local function checksInterconnectedNetwork2(target)
+	local countTarget = 0
+	local distance = {}
+	local keyRouts = {}
+	local loopRoute = true
+
+	while target[countTarget] do
+		local line = target[countTarget]
+		local countline = 1
+		local bePoint = getBELine(line)
+		local loopLine = {}
+		local loopLineCounte = 2
+
+		table.insert(loopLine, line)
+		keyRouts[line] = getKey(line)
+
+		distance[line.OBJET_ID_8.."P1"] = bePoint.p1:distance(line.targetPoint)
+		distance[line.OBJET_ID_8.."P2"] = bePoint.p2:distance(line.targetPoint)
+
+		while loopRoute do
+			local fsPointers
+
+			if line.targetPoint ~= nil then
+				fsPointers = checkLinePoints(line.route[countline], line.targetPoint)
+			else
+				fsPointers = checkLinePoints(line.route[countline], line.targ)
+			end
+
+			keyRouts[line.route[countline]] = distanceRoute(line, line.route[countline], distance, fsPointers)
+
+			countline = countline + 1
+
+			if not line.route[countline] then
+
+				getNextRoute(line, loopLine)
+
+				if loopLine[loopLineCounte] then
+					line = loopLine[loopLineCounte]
+					line.targ = fsPointers.secondP
+				end
+
+				loopLineCounte = loopLineCounte + 1
+				countline = 1
+
+				if not loopLine[loopLineCounte] then
+					loopRoute = false
+				end
+			end
+		end
+
+		countTarget = countTarget + 1
+	end
+
+	local network = {
+		keys = keyRouts,
+		distance = distance
+	}
+
+	return network
+end
+
+local function createOpenNetwork(self)
+	if self.lines.geometry then
```

Comment:
This verification should not be done here, or it should have an `else` stopping the function with an error.